finetune/adapter.py

```diff
@@ -218,7 +218,8 @@ def validate(
     for k in range(eval_iters):
         input_ids, targets = get_batch(fabric, val_data, longest_seq_length)
         logits = model(input_ids)
-        loss = chunked_cross_entropy(logits, targets, chunk_size=0)
+        logits[-1] = logits[-1][..., :-1, :]
+        loss = chunked_cross_entropy(logits, targets[..., 1:], chunk_size=0)
```

Comment:
Should we be using `chunk_size=0` in the training loss calculation as well? It's currently unset there

Reply:
Ah ok sounds like we shouldn't — chunk_size=0 is an optimization for the backward pass? (https://github.com/Lightning-AI/lit-gpt/pull/178/files#r1236187459)

Reply:
Given this, shouldn't this `chunk_size=0` in `finetune.py`'s training loss calculation be removed? https://github.com/Lightning-AI/lit-gpt/blob/75bca51f7e86759f52ca258a9aee40922177de7f/finetune/full.py#L168

Reply:
chunk_size>0 only helps with memory peaks for models with frozen parameters, the one you linked doesn't freeze at all (hence the "full" name)